src/error.rs

```diff
@@ -85,6 +88,8 @@ impl<'tcx> fmt::Display for EvalError<'tcx> {
                 write!(f, "array index {} out of bounds {} at {:?}", index, len, span),
             EvalError::Math(span, ref err) =>
                 write!(f, "mathematical operation at {:?} failed with {:?}", span, err),
+            EvalError::InvalidChar(c) =>
+                write!(f, "invalid utf8 character: {}", c),
```

Comment:
To be fully pedantic, it would be an invalid "Unicode Scalar Value" since that's how Rust's `char` is defined.

I'd rather use the message from `description`:

```rust
write!(f, "tried to interpret an invalid 32-bit value as a char: {}", c),
```